compiler/lib/fs.mli

```diff
@@ -21,3 +21,9 @@ val find_in_path : string list -> string -> string option
 val absolute_path : string -> string
 
 val read_file : string -> string
+
+val write_file : name:string -> contents:string -> unit
+
+val gen_file : string -> (string -> 'a) -> 'a
+
+val with_intermediate_file : string -> (string -> 'a) -> 'a
```

Comment:
What is this function doing ?  I don't see any intermediate_file in the implementation. How is it used ? Maybe the name is misleading ?

Reply:
We are calling external tools (the Binaryen toolchain) and use temporary intermediate files for that.

Function `with_intermediate_file` is used to make sure that a temporary file is deleted in case of failure. It is similar to `Filename.with_open` but passes the temporary file name instead of a channel. I use it this way:
```ocaml
Fs.with_intermediate_file (Filename.temp_file "wasm-merged" ".wasm")
@@ fun temp_file ->
...
```